### ISSUE TITLE
Error when installing (upgrading) existing (non ceedling) project 

### ISSUE THREAD

**Issue by automat01** [ThrowTheSwitch/Ceedling#703]
First of all, this is my first post on github.

We use many existing projects on which we wish to add unit tests using ceedling. our approach is to add the required files manually (files that are normally generated by ceedling using `new`). But we have an annoying issue if the `project_x/test/support` folder does not exist.  
In some case we need to have tests in a custom folder. I have found on my ceedling 0.31.1 instance this annoying line:
```
# Genarate gitkeep in test support path
        FileUtils.touch(File.join(test_support_path, '.gitkeep'))
```
where `test_support_path` is hardcoded as `test_support_path = File.join(name, 'test/support')` a few lines above.

Why is the script failing when it can't `touch` this?  
If it is absolutely necessary, is there a possibility to make this a part of the yml configuration? I don't think ceedling need to hardcode the `test/` folder. tests can be anywhere as per the `project.yml` file.

**Comment by lukzeg** [comment]
I was able to reproduce this issue, let me update a little bit description:

---
**Description**

Upgrading existing project without existing <project>/test/support folder cause build failure:

```
/usr/lib/ruby/3.0.0/fileutils.rb:1150:in `initialize': No such file or directory @ rb_sysopen - test_gcc_compile/test/support/.gitkeep (Errno::ENOENT)
	from /usr/lib/ruby/3.0.0/fileutils.rb:1150:in `open'
	from /usr/lib/ruby/3.0.0/fileutils.rb:1150:in `rescue in block in touch'
	from /usr/lib/ruby/3.0.0/fileutils.rb:1146:in `block in touch'
	from /usr/lib/ruby/3.0.0/fileutils.rb:1144:in `each'
	from /usr/lib/ruby/3.0.0/fileutils.rb:1144:in `touch'
	from /var/lib/gems/3.0.0/gems/ceedling-0.31.1/bin/ceedling:98:in `copy_assets_and_create_structure'

```

---

**Steps to reproduce**

1. In existing project folder delete test/support folder.
2. Run:  ```ceedling upgrade **<project_name>**```

---

**Actual Result:**

```
/usr/lib/ruby/3.0.0/fileutils.rb:1150:in `initialize': No such file or directory @ rb_sysopen - test_gcc_compile/test/support/.gitkeep (Errno::ENOENT)
	from /usr/lib/ruby/3.0.0/fileutils.rb:1150:in `open'
	from /usr/lib/ruby/3.0.0/fileutils.rb:1150:in `rescue in block in touch'
	from /usr/lib/ruby/3.0.0/fileutils.rb:1146:in `block in touch'
	from /usr/lib/ruby/3.0.0/fileutils.rb:1144:in `each'
	from /usr/lib/ruby/3.0.0/fileutils.rb:1144:in `touch'
	from /var/lib/gems/3.0.0/gems/ceedling-0.31.1/bin/ceedling:98:in `copy_assets_and_create_structure'

```

---

**Expected Result:**

Unknown behaviour.

---

Hi @mvandervoord ,

Should Ceedling try to apply default settings at default. Try to create test/support folder or as it will feel more understandable:
- load project.yml and check if the ```
:paths:
  :supports:
``` key exists and apply some changes. In case if not exists, apply default configuration( that means create <project_name>/test/support folder and create .gitkeep folder inside it).
- if the support key exists but point to another folder, the default location defined in https://github.com/ThrowTheSwitch/Ceedling/blob/master/bin/ceedling#L81 should be updated to defined in project.yml.

What is your opinion, how Ceedling should threat such of situation?

**Comment by lukzeg** [comment]
Hi @automat01  ,

Mentioned by you issue should be fixed in PR #738. If you will have time, can you check master branch and if fully solve your issue, can we close this this issue?

Kind Regards,
Lucas

**Comment by automat01** [comment]
Hi @lukzeg ,

Unfortunately , setting up to run ceedling from repo would require too much time for me to justify. You have described the problem really well and I have looked at #738 I think it should be fixed. I will definitely be checking out 0.32 release!

Thanks for your time.
Mathieu
